src/m_menu.c

```diff
@@ -2931,138 +2772,117 @@ setup_menu_t keys_settings5[] =  // Key Binding screen strings
 
-  {"HELP"        ,S_SKIP|S_KEEP ,m_scrn,0   ,0    ,{&key_help}},
-  {"MENU"        ,S_SKIP|S_KEEP ,m_scrn,0   ,0    ,{&key_escape}},
-  {"PAUSE"       ,S_INPUT     ,m_scrn,KB_X,M_Y+ 1*M_SPC,{0},input_pause},
-  {"SAVE"        ,S_INPUT     ,m_scrn,KB_X,M_Y+ 2*M_SPC,{0},input_savegame},
-  {"QUICKSAVE"   ,S_INPUT     ,m_scrn,KB_X,M_Y+ 3*M_SPC,{0},input_quicksave},
-  {"QUICKLOAD"   ,S_INPUT     ,m_scrn,KB_X,M_Y+ 4*M_SPC,{0},input_quickload},
-  {"LOAD"        ,S_INPUT     ,m_scrn,KB_X,M_Y+ 5*M_SPC,{0},input_loadgame},
-  {"VOLUME"      ,S_INPUT     ,m_scrn,KB_X,M_Y+ 6*M_SPC,{0},input_soundvolume},
-  {"HUD"         ,S_INPUT     ,m_scrn,KB_X,M_Y+ 7*M_SPC,{0},input_hud},
-  {"END GAME"    ,S_INPUT     ,m_scrn,KB_X,M_Y+ 8*M_SPC,{0},input_endgame},
-  {"MESSAGES"    ,S_INPUT     ,m_scrn,KB_X,M_Y+ 9*M_SPC,{0},input_messages},
-  {"QUIT"        ,S_INPUT     ,m_scrn,KB_X,M_Y+10*M_SPC,{0},input_quit},
-  {"GAMMA FIX"   ,S_INPUT     ,m_scrn,KB_X,M_Y+11*M_SPC,{0},input_gamma},
-  {"SPY"         ,S_INPUT     ,m_scrn,KB_X,M_Y+12*M_SPC,{0},input_spy},
-  {"LARGER VIEW" ,S_INPUT     ,m_scrn,KB_X,M_Y+13*M_SPC,{0},input_zoomin},
-  {"SMALLER VIEW",S_INPUT     ,m_scrn,KB_X,M_Y+14*M_SPC,{0},input_zoomout},
-  {"SCREENSHOT"  ,S_INPUT     ,m_scrn,KB_X,M_Y+15*M_SPC,{0},input_screenshot},
-  {"CLEAN SCREENSHOT",S_INPUT ,m_scrn,KB_X,M_Y+16*M_SPC,{0},input_clean_screenshot},
-
-  {"<- PREV", S_SKIP|S_PREV,m_null,M_X_PREV,M_Y_PREVNEXT, {keys_settings4}},
-  {"NEXT ->", S_SKIP|S_NEXT,m_null,M_X_NEXT,M_Y_PREVNEXT, {keys_settings6}},
+  {"Help",       S_SKIP|S_KEEP, m_scrn, 0,    0,     {&key_help}},
+  {"Menu",       S_SKIP|S_KEEP, m_scrn, 0,    0,     {&key_escape}},
+  {"Pause",            S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_pause},
+  {"Save",             S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_savegame},
+  {"Quicksave",        S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_quicksave},
+  {"Quickload",        S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_quickload},
+  {"Load",             S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_loadgame},
+  {"Volume",           S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_soundvolume},
+  {"Hud",              S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_hud},
+  {"End Game",         S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_endgame},
+  {"Messages",         S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_messages},
+  {"Quit",             S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_quit},
+  {"Gamma Fix",        S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_gamma},
+  {"Spy",              S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_spy},
+  {"Larger View",      S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_zoomin},
+  {"Smaller View",     S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_zoomout},
+  {"Screenshot",       S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_screenshot},
+  {"Clean Screenshot", S_INPUT, m_scrn, KB_X, M_SPC, {0}, input_clean_screenshot},
+
+  {"<- PREV", S_SKIP|S_PREV, m_null, M_X_PREV, M_Y_PREVNEXT, {keys_settings3}},
```

Comment:
Could you factor out PREV/NEXT/RESET menu entries into macros?

Reply:
Since there are no longer direct coordinates for menu items, we can turn many things into macros, like this:
```c
{
   M_TITLE,
   M_THERMO12(resolution_scale),
   M_YESNO(widesreen),
   M_GAP,
   M_NEXT(gen_settings2),
   M_END
}
```
I tried it, but I didn't like it. Sure, less clutter, but after a while it will be hard to remember what's going on.